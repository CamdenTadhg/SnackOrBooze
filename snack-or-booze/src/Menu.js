import React from "react";
import { Link } from "react-router-dom";
import "./Menu.css";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  ListGroup,
  ListGroupItem
} from "reactstrap";

function Menu({ items, type }) {
  const capitalize = (string) => {
    return string.charAt(0).toUpperCase() + string.slice(1);
  }

  return (
    <section className="col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            {capitalize(type)} Menu
          </CardTitle>
          <CardText>
            Our delicious {type}
          </CardText>
          <ListGroup>
            {items.map(item => (
              <Link to={`/${type}/${item.id}`} key={item.id}>
                <ListGroupItem>{item.name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
        </CardBody>
      </Card>
    </section>
  );
}

export default Menu;
