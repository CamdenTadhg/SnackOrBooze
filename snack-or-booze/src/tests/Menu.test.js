import React from 'react';
import {render} from '@testing-library/react';
import Menu from "../Menu";

const items = [    {
  "id": "nachos",
  "name": "Nachos",
  "description": "An American classic!",
  "recipe": "Cover expensive, organic tortilla chips with Cheez Whiz.",
  "serve": "Serve in a hand-thrown ceramic bowl, garnished with canned black olives"
},
{
  "id": "hummus",
  "name": "Hummus",
  "description": "Sure to impress your vegan friends!",
  "recipe": "Purchase one container of hummus.",
  "serve": "Place unceremoniously on the table, along with pita bread."
}];

it("renders without crashing", function() {
    render(<Menu items={items} type='snacks'/>);
});

it("matches snapshot", function() {
    const {asFragment} = render(<Menu items={items} type='snacks'/>);
    expect(asFragment()).toMatchSnapshot();
});

it('displays the correct content', function() {

    
    const {getByText} = render(<Menu items={items} type="snacks"/>);
    expect(getByText('Snack Menu', {exact: false})).toBeInTheDocument();
    expect(getByText('Hummus', {exact: false})).toBeInTheDocument();
});