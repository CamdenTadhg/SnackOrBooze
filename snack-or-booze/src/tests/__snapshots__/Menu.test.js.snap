// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          Snacks Menu
        </h5>
        <p
          class="card-text"
        >
          Our delicious snacks
        </p>
        <ul
          class="list-group"
        >
          <a
            href="/snacks/nachos"
          >
            <li
              class="list-group-item"
            >
              Nachos
            </li>
          </a>
          <a
            href="/snacks/hummus"
          >
            <li
              class="list-group-item"
            >
              Hummus
            </li>
          </a>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
